fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,624 @@
+| `COMMITTEE_WEIGHT_ESTIMATION_ADJUSTMENT_FACTOR` | `int(5)`  | Per mille value to add to the estimation of the committee weight across a range of slots not covering a full epoch in order to ensure the safety of the confirmation rule with high probability. See [here](https://gist.github.com/saltiniroberto/9ee53d29c33878d79417abb2b4468c20) for an explanation about the value chosen. |
+
+### Configuration
+
+The confirmation rule can be configured to the desired tolerance of Byzantine validators, for which the algorithm takes the following input parameters:
+
+| Input Parameter                    | Type     | Max. Value               | Description                                                                                             |
+|------------------------------------|----------|:-------------------------|---------------------------------------------------------------------------------------------------------|
+| `CONFIRMATION_BYZANTINE_THRESHOLD` | `uint64` | `MAX_CONFIRMATION_SCORE` | assumed maximum percentage of Byzantine validators among the validator set.                             |
+| `CONFIRMATION_SLASHING_THRESHOLD`  | `Gwei`   | `2**64 - 1`              | assumed maximum amount of stake that the adversary is willing to get slashed in order to reorg a block. |
+
+
+### Helper Functions
+
+### `committee_spans_full_epoch`
+
+```python
+def committee_spans_full_epoch(start_slot: Slot, end_slot: Slot) -> bool:
+    """
+    Returns whether the range from ``start_slot`` to ``end_slot`` (inclusive of both) includes and entire epoch
+    """
+    start_epoch = compute_epoch_at_slot(start_slot)
+    end_epoch = compute_epoch_at_slot(end_slot)
+
+    return (
+        end_epoch > start_epoch + 1 or
+        (end_epoch == start_epoch + 1 and start_slot % SLOTS_PER_EPOCH == 0))
+```
+
+### `committee_for_block_spans_full_epoch`
+
+```python
+def committee_for_block_spans_full_epoch(store: Store, block_root: Root) -> bool:
+    """
+    Returns whether the range from ``start_slot`` to ``end_slot`` (inclusive of both) includes and entire epoch
+    """
+    current_slot = get_current_slot(store)
+    block = store.blocks[block_root]
+    parent_block = store.blocks[block.parent_root]
+    
+    return committee_spans_full_epoch(Slot(parent_block.slot + 1), current_slot)
+```
+
+### `ceil_div`
+
+```python
+def ceil_div(numerator: int, denominator: int) -> int:
+    """
+    Returns ``ceil(numerator / denominator)`` using only integer arithmetic
+    """
+    result = numerator // denominator
+    if numerator % denominator != 0:
+        result = result + 1
+    return result
```

Comment:
Alternative version:
```python
def ceil_div(numerator: int, denominator: int) -> int:
    return ((numerator - 1) // denominator) + 1
```

Reply:
I think that the orginal version is better for a spec as it better clarifies the expect output of the fuction.

Reply:
Changed to your alternative @mkalinin as I think that it is more readable.